src/wdio/services/SeleniumDockerService.js

```diff
@@ -65,39 +31,130 @@ export default class SeleniumDockerService {
     this.port = config.port;
     this.path = config.path;
     this.browserName = capabilities[0].browserName;
-    this.cidfile = this.config.cidfile || this.cidfile;
 
+    if (!this.config.enabled) {
+      return;
+    }
+
+    this.container = await this.getRunningContainer();
+
+    // There is a running container that doesn't match configuration, stop it
+    // before proceeding
+    if (this.container && this.getImage() !== this.container.image) {
+      await this.stop();
+
+    // There is a container, verify selenium is running as expected
+    } else if (this.container) {
+      try {
+        await this.ensureSelenium();
+
+      // Selenium isn't responding stop the container;
+      } catch (e) {
+        await this.stop();
+      }
+    }
+
+    if (!this.container) {
+      await this.pullImage();
+      await this.runImage();
+      await this.ensureSelenium();
+    }
+  }
+
+  /**
+   * Clean up docker container after all workers got shut down and the process is about to exit.
+   */
+  async onComplete() {
+    if (this.config.cleanup && this.config.enabled) {
+      await this.stop();
+    }
+  }
+
+  /**
+  * Waits for selenium to startup and be ready within the configured container.
+  * @return {Promise}
+  */
+  ensureSelenium() {
     return new Promise((resolve, reject) => {
-      if (this.config.enabled) {
-        const containerId = this.getContainerId();
-
-        // Workaround for browser specific docker issues
-        // see https://github.com/SeleniumHQ/docker-selenium#running-the-images
-        let args = '';
-        if (this.browserName === 'chrome') {
-          args = '-v /dev/shm:/dev/shm';
-        } else if (this.browserName === 'firefox') {
-          args = '--shm-size 2g';
+      console.log('Ensuring selenium status is ready');
+      // Retry for 500 times up to 5 seconds for selenium to start
```

Comment:
Update inline to 10 seconds

Reply:
i should probably just remove that comment.. but currenlty it only waits 5 seconds, do you want that upped to 10?

Reply:
The default retry interval above is set to 10 with retries at 500, but maybe just update the inline to indicate its waiting for selenium to start. 

I don't have an opinion on the wait/check times. I'm not sure what would be ideal, though upping it wouldn't necessarily hurt if its completes before the timeout it'll just move on.